### PR TITLE
fix: connect to sharded redis cluster (#82)

### PR DESCRIPTION
**Pull request by dcompton-eg**
* allow pre-configured client

### CODE REVIEW

**Review by mcollina — CHANGES_REQUESTED** [review]
Can you add a test for this?

**Comment by dcompton-eg** [comment]
I've added a test for the external client connection based on existing tests and [coverage](https://github.com/moscajs/aedes-persistence-redis/blob/master/test.js#L18) in aedes-persistence-redis.

File: mqemitter-redis.js [hunk 1]
```diff
@@ -18,8 +18,13 @@ function MQEmitterRedis (opts) {
 
   this._opts = opts
 
-  this.subConn = new Redis(opts.connectionString || opts)
-  this.pubConn = new Redis(opts.connectionString || opts)
+  if (opts.cluster && Array.isArray(opts.cluster)) {
```
Comment (mcollina) [inline mqemitter-redis.js:21]: I don't see how this branch is triggered

Comment (dcompton-eg) [reply mqemitter-redis.js:21]: Would you like to see an integration test with a redis cluster in the workflow?   If so, would you be open to reducing scope to only allow for external connection?

Comment (mcollina) [reply mqemitter-redis.js:21]: Yes

**Review by mcollina — APPROVED** [review]
lgtm

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**